plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,463 @@
+	case "READ", "WRITE":
+		fields["ops"] = nline[0]
+		fields["retrans"] = nline[1] - nline[0]
+		fields["bytes"] = nline[3] + nline[4]
+		fields["rtt"] = nline[6]
+		fields["exe"] = nline[7]
+		tags["operation"] = first
+		acc.AddFields("nfsstat", fields, tags)
+	}
+
+	if fullstat {
+		switch first {
+		case "events":
+			if len(nline) >= len(eventsFields) {
+				for i, t := range eventsFields {
+					fields[t] = nline[i]
+				}
+				acc.AddFields("nfs_events", fields, tags)
+			}
+
+		case "bytes":
+			if len(nline) >= len(bytesFields) {
+				for i, t := range bytesFields {
+					fields[t] = nline[i]
+				}
+				acc.AddFields("nfs_bytes", fields, tags)
+			}
+
+		case "xprt":
+			if len(line) > 1 {
+				switch line[1] {
+				case "tcp":
+					if len(nline)+2 >= len(xprttcpFields) {
+						for i, t := range xprttcpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_tcp", fields, tags)
+					}
+				case "udp":
+					if len(nline)+2 >= len(xprtudpFields) {
+						for i, t := range xprtudpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_udp", fields, tags)
+					}
+				}
+			}
+		}
+
+		if (version == "3" && n.nfs3Ops[first]) || (version == "4" && n.nfs4Ops[first]) {
+			tags["operation"] = first
+			if len(nline) <= len(nfsopFields) {
+				for i, t := range nline {
+					fields[nfsopFields[i]] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+func (n *NFSClient) processText(scanner *bufio.Scanner, acc telegraf.Accumulator) error {
+	var mount string
+	var version string
+	var export string
+	var skip bool
+
+	for scanner.Scan() {
+		line := strings.Fields(scanner.Text())
+
+		line_len := len(line)
+
+		if line_len == 0 {
+			continue
+		}
+
+		if line_len > 4 && choice.Contains("fstype", line) && (choice.Contains("nfs", line) || choice.Contains("nfs4", line)) {
+			mount = line[4]
+			export = line[1]
+			skip = false
+		} else if line_len > 5 && (choice.Contains("(nfs)", line) || choice.Contains("(nfs4)", line)) {
+			version = strings.Split(line[5], "/")[1]
+		}
+
+		if len(n.IncludeMounts) > 0 {
+			skip = true
+			for _, RE := range n.IncludeMounts {
+				matched, _ := regexp.MatchString(RE, mount)
+				if matched {
+					skip = false
+					break
+				}
+			}
+		}
+
+		if !skip && len(n.ExcludeMounts) > 0 {
+			for _, RE := range n.ExcludeMounts {
+				matched, _ := regexp.MatchString(RE, mount)
+				if matched {
+					skip = true
+					return nil // last test, and we're skipping anyway
+				}
+			}
+		}
+
+		if !skip {
+			n.parseStat(mount, export, version, line, n.Fullstat, acc)
+		}
+	}
+	return nil
+}
+
+func (n *NFSClient) getMountStatsPath() string {
+
+	path := "/proc/self/mountstats"
+	if os.Getenv("MOUNT_PROC") != "" {
+		path = os.Getenv("MOUNT_PROC")
+	}
+	n.Log.Debugf("Opening [%s] for mountstats", path)
+	return path
+}
+
+func (n *NFSClient) Gather(acc telegraf.Accumulator) error {
+
+	file, err := os.Open(n.getMountStatsPath())
```

Comment:
Please call `n.getMountStatsPath()` in `Init()` and store the result in the `NFSClient` struct.

Reply:
Moved.